src/components/Personalization/createPersonalizationDetails.js

```diff
@@ -20,8 +20,19 @@ import {
   REDIRECT_ITEM
 } from "./constants/schema";
 import isNonEmptyString from "../../utils/isNonEmptyString";
+import { WEB_SURFACE } from "./constants/surface";
+import getPageLocation from "./utils/getPageLocation";
 
-export default ({ renderDecisions, decisionScopes, event, viewCache }) => {
+const buildPageSurface = window => WEB_SURFACE + getPageLocation(window);
```

Comment:
Rather than passing window everywhere, you could pass this function that was already injected with window. i.e.
`components/Personalization/createBuildPageSurface.js`

```
import { WEB_SURFACE } from "./constants/surface";

export default ({ window }) => () => {
  const url = win.location.href;
  return WEB_SURFACE + url.substring(url.indexOf(":") + 3);
};
```

Reply:
Fixed